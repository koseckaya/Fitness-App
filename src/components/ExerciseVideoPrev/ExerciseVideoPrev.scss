@import "src/styles/vars";

.exercise-video-prev {
  box-sizing: border-box;
   margin-bottom: 10px;
    border-radius: 5px;

  &__img {
    object-fit: cover;
    margin-bottom: 10px;
    border-radius: 5px;
    width: 100%;
  }
}
