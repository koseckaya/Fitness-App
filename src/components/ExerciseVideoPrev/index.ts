export { default as ExerciseVideoPrev } from './ExerciseVideoPrev';
export type { Props as ExerciseVideoPrevProps } from './ExerciseVideoPrev';
