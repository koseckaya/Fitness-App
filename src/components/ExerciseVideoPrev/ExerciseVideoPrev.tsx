import { FC } from 'react';
import './ExerciseVideoPrev.scss';

export type Props = {
    title: string;
    src: string;
    srcImg: string;
};

const ExerciseVideoPrev: FC<Props> = ({title, src, srcImg}: Props) => {
        return (
            <div className='exercise-video-prev'>
                <a href={ src }>
                    <img className='exercise-video-prev__img' src={srcImg} alt={ title} />
                    <div>{title}</div>
                </a>
            </div>
    )
}
   
export default ExerciseVideoPrev;
