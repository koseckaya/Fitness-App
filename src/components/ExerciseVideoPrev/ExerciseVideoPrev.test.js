import React from 'react';
import renderer from 'react-test-renderer';
import  ExerciseVideoPrev from './ExerciseVideoPrev';


const props = {
    title: '5 Min Warm Up',
    src: 'https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing',
    srcImg: 'https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg',
    day: '1',
    active: false,
    onVideoClick: () => { },
};



describe("ExerciseVideoPrev component", () => {
    it('should render component with default props', () => {
        const component = renderer
            .create(<ExerciseVideoPrev {...props}></ExerciseVideoPrev>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render component without prop title', () => {
        const newProps = { ...props, title: '' }
        const component = renderer
            .create(<ExerciseVideoPrev {...newProps}></ExerciseVideoPrev>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without src prop', () => {
      
        const newProps = { ...props, src: '' }
        const component = renderer
            .create(<ExerciseVideoPrev {...newProps}></ExerciseVideoPrev>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without srcImg prop', () => {
        const newProps = { ...props, srcImg: '' }
        const component = renderer
            .create(<ExerciseVideoPrev {...newProps}></ExerciseVideoPrev>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component with inactive prop', () => {
        const newProps = { ...props, active: true }
        const component = renderer
            .create(<ExerciseVideoPrev {...newProps}></ExerciseVideoPrev>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})
