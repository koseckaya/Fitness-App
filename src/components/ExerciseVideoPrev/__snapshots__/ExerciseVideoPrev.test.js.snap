// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseVideoPrev component should render component with default props 1`] = `
<div
  className="exercise-video-prev"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="exercise-video-prev__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="video-prev "
      onClick={[Function]}
    />
    <div>
      5 Min Warm Up
    </div>
  </a>
</div>
`;

exports[`ExerciseVideoPrev component should render component with inactive prop 1`] = `
<div
  className="exercise-video-prev"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="exercise-video-prev__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="video-prev active"
      onClick={[Function]}
    />
    <div>
      5 Min Warm Up
    </div>
  </a>
</div>
`;

exports[`ExerciseVideoPrev component should render component without prop title 1`] = `
<div
  className="exercise-video-prev"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt=""
      className="exercise-video-prev__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="video-prev "
      onClick={[Function]}
    />
    <div />
  </a>
</div>
`;

exports[`ExerciseVideoPrev component should render component without src prop 1`] = `
<div
  className="exercise-video-prev"
>
  <a
    href=""
  >
    <img
      alt="5 Min Warm Up"
      className="exercise-video-prev__img"
      src="https://static.chloeting.com/videos/63ae100649cf64962cb594e2/f081bd90-8a98-11ed-97ac-4b900c40c009.jpeg"
    />
    <div
      className="video-prev "
      onClick={[Function]}
    />
    <div>
      5 Min Warm Up
    </div>
  </a>
</div>
`;

exports[`ExerciseVideoPrev component should render component without srcImg prop 1`] = `
<div
  className="exercise-video-prev"
>
  <a
    href="https://www.youtube.com/watch?v=7t1qhpDLSi0&ab_channel=ChloeTing"
  >
    <img
      alt="5 Min Warm Up"
      className="exercise-video-prev__img"
      src=""
    />
    <div
      className="video-prev "
      onClick={[Function]}
    />
    <div>
      5 Min Warm Up
    </div>
  </a>
</div>
`;
